src/malli/registry.cljc

```diff
@@ -8,11 +9,18 @@
   (-schema [this type] "returns the schema from a registry")
   (-schemas [this] "returns all schemas from a registry"))
 
-(defn simple-registry [schemas]
+(defn fast-registry [m]
+  (let [fm #?(:clj (doto (HashMap.) (.putAll ^Map m)), :cljs m)]
```

Comment:
Might want to tune the initial capacity and collision factor for the map. Since you want to minimize hash collisions, I found initial values like 1024 / 0.25 give better results

Reply:
thanks. Did you find any difference in using the IndentityMap?

Reply:
Nothing conclusive, sadly